App.js

```diff
@@ -231,51 +253,46 @@ const navTheme = {
   },
 };
 
+const NavigatorNameContext = React.createContext();
+ 
 export default class App extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
-      initialState: null,
-    };
+      Stack: checkIsSmallScreen(Dimensions.get('window').width) ? NativeStack : WebStack,
+    }
   }
+  
+  regenerateNavigationState() {
+      const currentState = navigationRef.getState()
+      const strippedState = stripNavigationState(currentState) 
+      addChatRouteIfNecessary(strippedState)
+      navigationRef.reset(strippedState)
+  }
+  
+  handleResize(e) {
+    const isSmallScreenWidth = checkIsSmallScreen(e.window.width);
+    const navigatorName = this.state.Stack.Navigator.name;
 
-  componentDidMount() {
-    Linking.getInitialURL().then((url) => {
-      if (Platform.OS !== 'web') {
-        this.setState({initialState: getStateFromPath('', config)});
-        return;
-      }
-
-      // From here we'll have a url and we need to build the initial state for the app
-      // This is where things get tricky because if we have a report route then we need to parse the reportID and also make sure a LHN is the root view
-      const pathname = new URL(url).pathname;
-      const state = getStateFromPath(pathname, config);
-
-      // If pathname is a chat then we need to push a left hand nav path
-      if (pathname.startsWith('/r') || pathname.startsWith('/search') || pathname.startsWith('/settings')) {
-        state.routes.unshift({name: 'LeftHandNav'});
-      }
-
-      // Since About is in a Settings stack we want the "up" button to go back to the Settings main page and so need to define this
-      if (pathname === '/settings/about') {
-        state.routes[1].state.routes.unshift({name: 'Settings'});
-      }
-
-      // Add a report when we are on large format web and a chat does not exist in the route
-      if (!isSmallScreenWidth && !state.routes.find(route => route.name === 'Chat')) {
-          state.routes.splice(1, 0, {name: 'Chat', params: {id: 1}});
-      }
```

Comment:
I am not sure if the `initialState` is necessary? It looks like we can handle this just with the right linking config

Reply:
I'm not sure if the config can be used to do everything (but maybe there's a way to also make it aware of the screen dimensions).

One breaking change here is that if we navigate directly to the site root on mobile web we are directed to a chat and the expected behavior would be to remain on the sidebar.